### ISSUE TITLE
Add support for XZ compression

### ISSUE THREAD

**Issue by claunia** [uwedeportivo/romba#108]
It will provide less depot size than gzip.

**Comment by Obiwantje** [comment]
gz was initially chosen as a good balance between space and speed. Have you tried to take a depot and convert that from gz to xz to see what savings/time this would look like all?

**Comment by claunia** [comment]
I don't think timing is so much of an importance in an archiving situation.

But I'll give you results asap with a 33Gb test depot.

**Comment by Obiwantje** [comment]
Claunia - 30GB is way too small of a depot to give any representative results - even MAME is larger. Let's try a test with a minimum of say 500GB with mixed content/size (redump, MAME, some TOSEC?)

**Comment by claunia** [comment]
@Obiwantje I can't prepare 500Gb of free disc space for the test, but the results will be the same, as is common knowledge, LZMA (xz) always gets better compression than Deflate (gzip).

Nonetheless, as xz manpage says:
````
-0 is sometimes faster than gzip -9 while compressing much better.  The higher ones often have speed comparable to bzip2(1) with comparable or better compression ratio, although the results depend a lot on the type of data being compressed.
````

And I can assert that is true, with the 33Gb set I prepared, xz -0 was as fast as (mostly slowed by being 63k different files instead of a single one) gzip, getting a depot size 5% smaller.

Getting even smaller files (as much as 50% smaller than gzip in some sets I tested months ago from WinWorldPC) is just an issue of available RAM and time, something that the enduser can configure.

Finally, gzip is single-threaded while xz is already multi-threaded for compression.

Also, my test only included Atari, so if you include everything you say, you'll get even bigger differences, as the big files of TOSEC-ISO benefit a lot from LZMA's dictionary sizes.

Btw, MAME CHD are already using LZMA.
